docs/decisions/0013-library-reference-content-block.rst

```diff
@@ -0,0 +1,61 @@
+Referencing Content Blocks in Library V2
+--------------------------------------------------
+
+Status
+=======
+Pending
+
+Context
+=======
+Currently, the courses are being stored in the Modulestore and the libraries are stored in both Modulestore and the Blockstore. We want the library content to be stored in the Blockstore.
+We want to have a content reuse mechanism that allows authors to incorporate content (XBlocks) from content libraries into their courses without making a copy of it; in other words, we want a referencing mechanism for library content. 
+
+We are currently building out the functionality of libraries v2, and trying to migrate all usage of content libraries onto Blockstore.
+The benefits of using the Blockstore backed v2 libraries are:
+
+#. Blockstore is much more flexible than modulestore and can store anything that can be represented as a file - unlike modulestore, which only stores courses built of XBlocks. Also, the Blockstore is being moved into the edx-platform.
+#. simpler to maintain than modulestore
+#. increases re-usability of the content
+
+Terminology Used
+^^^^^^^^^^^^^^^^
+#. Blocks: Each reusable piece of content in a library. Example: problems, videos, HTML. Also known as Xblocks.
+#. Library: Collection of blocks in a course.
+
+
+Decisions
+=========
+We want to be able to reference single and multiple specific blocks from a library (v2, as well as v1) into a course (that are stored in modulestore)
+The workflow discussed for the interface for library content referencing would be:
+
+#. Author chooses a library.
+#. Author chooses a pool of blocks.
+#. If multiple blocks of the same type are selected, ask for randomization
+#. If no randomization, then ask for ordering
```

Comment:
I see now that perhaps the randomization workflow being included here is because we are building / planning to have both a Library Randnomize and Library Reference block. It might be helpful to separate these two as different workflows with their own steps, even if the implementation behind the scenes has much of the same shared code / infrastructure to enable this?

Reply:
Understood

Reply:
This part still needs to be split into two workflows, but also it's perhaps not necessary to say the details of the UX workflow here. We can just delete this numbered list of steps and say "Studio will provide authors with two separate workflows, once for selecting one or more specific blocks and adding those specific blocks to the course, and another workflow for selecting a randomized set of blocks, so that each student will see a random block out of the set selected by the author"